### ISSUE TITLE
Incorrect version numbers for feature branch builds

### ISSUE THREAD

**Issue by donmesserli** [GitTools/GitVersion#2066]
I've just started working at a company and I'm being told that this used to work.
We are using Azure DevOps and the version of GitVersion is 5.0.1
We do NOT have a GitVersionConfig.yaml or a GitVersion.yml file.

Our workflow is as follows...
Start with Master
Create a Release branch from Master (i.e. release/1.14)
Create a Feature branch from the Release branch (i.e. feature/AAAA-123)
A build is automatically kicked off and the version ends up being from a previous Release (i.e. 1.12.0)+75 (6000)
We need the version to be 1.14.0+xx (6000)
When a pull request of feature/AAAA-123 into release/1.14 a release build is performed and it gets an incorrect version number
Then a rolling build of the release is performed and it gets the correct version number (1.14.0+xx (xxxx).

I'm told this used to work. What are we doing wrong?

**Comment by asbjornu** [comment]
I'm not 100% sure, but this seems like a duplicate of #2056, or at least closely related to it.

**Comment by donmesserli** [comment]
I've added a GitVersion.yml file. It is as follows...
`assembly-versioning-scheme: MajorMinorPatchTag
tag-prefix: '[vVrRsS]'
branches:
 master:
  regex: master
  tag: ci
  increment: None
 feature:
  regex: features?[/-]
  increment: None
 release:
  regex: releases?[/-]
  tag: ''
  increment: None
 other:
  regex: (?!(master|release|feature)).+[/-]
  increment: Inherit
  source-branches: ['master', 'release', 'feature']
ignore:
 sha: []`

However, it only works if I've tagged the Release branch before making the Feature branch.

Any suggestions?
